hoomd/hpmc/ComputeSDF.h

```diff
@@ -364,13 +390,184 @@ template<class Shape> void ComputeSDF<Shape>::countHistogram(uint64_t timestep)
                     }
                 } // end loop over AABB nodes
             }     // end loop over images
+        if (min_bin_ptl_i < m_hist.size())
+            {
+            m_hist[min_bin_ptl_i]++;
+            }
+        } // end loop over all particles
+    }     // end countHistogram()
+
+template<class Shape> void ComputeSDF<Shape>::countHistogramBruteForce(uint64_t timestep)
+    {
+    // update the aabb tree
+    const hoomd::detail::AABBTree& aabb_tree = m_mc->buildAABBTree();
+
+    // update the image list
+    const std::vector<vec3<Scalar>>& image_list = m_mc->updateImageList();
+
+    Scalar extra_width = m_xmax / (1 - m_xmax) * m_mc->getMaxCoreDiameter();
+
+    // access particle data and system box
+    ArrayHandle<Scalar4> h_postype(m_pdata->getPositions(),
+                                   access_location::host,
+                                   access_mode::read);
+    ArrayHandle<Scalar4> h_orientation(m_pdata->getOrientationArray(),
+                                       access_location::host,
+                                       access_mode::read);
+    ArrayHandle<Scalar> h_diameter(m_pdata->getDiameters(),
+                                   access_location::host,
+                                   access_mode::read);
+    ArrayHandle<Scalar> h_charge(m_pdata->getCharges(), access_location::host, access_mode::read);
+    const std::vector<param_type, hoomd::detail::managed_allocator<param_type>>& params
+        = m_mc->getParams();
+
+    // loop through N particles
+    // At the top of this loop, we initialize min_bin_ptl_i to the size of the sdf histogram
+    // For each of particle i's neighbors, we find the scaling that produces the first overlap.
+    // For each neighbor, we do a brute force search from the scaling corresponding to bin 0
+    // up to the minimum bin that we've already found for particle i.
+    // Then we add to m_hist[min_bin_ptl_i] the negative Mayer-function corresponding to the type
+    // of overlap corresponding to particle i's first overlap.
+    for (unsigned int i = 0; i < m_pdata->getN(); i++)
+        {
+        size_t min_bin_ptl_i = m_hist.size();
+        double hist_weight_ptl_i = 2.0;
+
+        // read in the current position and orientation
+        Scalar4 postype_i = h_postype.data[i];
+        Scalar4 orientation_i = h_orientation.data[i];
+        int typ_i = __scalar_as_int(postype_i.w);
+        Shape shape_i(quat<Scalar>(orientation_i), params[__scalar_as_int(postype_i.w)]);
+        vec3<Scalar> pos_i = vec3<Scalar>(postype_i);
+
+        // TODO: do we need to account for patch r_cut in here somewhere?
```

Comment:
Yes, you need to account for the patch r_cut. If you don't, you will miss potentially interacting pairs of particles.
See `IntegratorHPMC` for details on how to include the r_cut into the AABB tree `R_query`.